workflows/sagemaker_pipelines/SPECIFICATION.md

```diff
@@ -0,0 +1,332 @@
+#  SageMaker workflow technical specification
+
+This document contains the technical specifications for the Amazon SageMaker Geospatial Pipelines example, a sample workflow that showcases SageMaker pipelines using SDKs.
+
+This document explains the following:
+
+- Application inputs and outputs
+- Underlying AWS components and their configurations
+- Implementation details and sample output
+- Troubleshooting information
+
+### Table of contents
+
+- [Architecture](#architecture)
+- [Common resources](#common-resources)
+- [Metadata](#metadata)
+- [Implementation](#implementation)
+- [Troubleshooting](#troubleshooting)
+
+## Architecture
+This workflow uses a pre-defined SageMaker pipeline to execute a geospatial job in SageMaker. The pipeline uses [Pipeline steps](https://docs.aws.amazon.com/sagemaker/latest/dg/build-and-manage-steps.html) to define the actions and relationships of the pipeline operations. The pipeline in this example includes an [AWS Lambda step](https://docs.aws.amazon.com/sagemaker/latest/dg/build-and-manage-steps.html#step-type-lambda)
+and a [callback step](https://docs.aws.amazon.com/sagemaker/latest/dg/build-and-manage-steps.html#step-type-callback).
+Both steps are processed by the same example Lambda function.
+
+The Lambda function handler should be written as part of the example, with the following functionality:
+- Starts a [SageMaker Vector Enrichment Job](https://docs.aws.amazon.com/sagemaker/latest/dg/geospatial-vej.html) with the provided job configuration.
+- Processes Amazon SQS queue messages from the SageMaker pipeline.
+- Starts the export function with the provided export configuration.
+- Completes the pipeline when the export is complete.
+
+This diagram represents the relationships between key components.
+![relational diagram](resources/workflow.png)
+
+AWS SageMaker is a managed machine learning service. Developers can build and train machine learning models and deploy them into a production-ready hosted environment. This example focuses on the pipeline capabilities rather than the model training and building capabilities, since those are more likely to be useful to an SDK developer.
```

Comment:
```suggestion
Amazon SageMaker is a managed machine learning service. Developers can build and train machine learning models and deploy them into a production-ready hosted environment. This example focuses on the pipeline capabilities rather than the model training and building capabilities, since those are more likely to be useful to an SDK developer.
```